scripts/examples/typescript/src/XenAPI.ts

```diff
@@ -0,0 +1,52 @@
+import { JSONRPCClient } from "json-rpc-2.0";
+
+class XapiSession {
+    private client: JSONRPCClient;
+    private session_id: string | undefined;
+
+    constructor(hostUrl: string | undefined) {
+        this.client = new JSONRPCClient(async (request) => {
+            const response = await fetch(`${hostUrl}/jsonrpc`, {
+                method: "POST",
+                headers: { "content-type": "application/json" },
+                body: JSON.stringify(request),
+            });
+            if (response.status === 200) {
```

Comment:
To add a bit to the above comment, we should check whether the response Json contains Error and if it does throw an exception containing the Message and Data, otherwise the user would have to do that for every call.